game.js

```diff
@@ -9,146 +9,369 @@ Here, we create and add our "canvas" to the page.
+    playerName !== "" &&
+    currentRemainingTime !== "0" &&
+    eggCount >= 0 &&
+    eggCount < maxEgg
+  ) {
+    update();
+    render();
+    let eggCountNum = document.getElementById("eggCountHTML");
+    eggCountNum.innerHTML = eggCount;
+  }
+
+  if (eggCount === maxEgg) {
+    win();
+  }
+
+  if (currentRemainingTime === "0") {
+    lose();
+  }
+
+  // ctx.font = "12px Comic Sans MS";
+  // ctx.fillStyle = "red";
+
+  // ctx.fillText(`Remaining time: ${remainTime}s`, 50, 40);
   // Request to do this again ASAP. This is a special method
-  // for web browsers. 
+  // for web browsers.
   requestAnimationFrame(main);
 };
 
 // Cross-browser support for requestAnimationFrame.
 // Safely ignore this line. It's mostly here for people with old web browsers.
 var w = window;
-requestAnimationFrame = w.requestAnimationFrame || w.webkitRequestAnimationFrame || w.msRequestAnimationFrame || w.mozRequestAnimationFrame;
+
+requestAnimationFrame =
+  w.requestAnimationFrame ||
+  w.webkitRequestAnimationFrame ||
+  w.msRequestAnimationFrame ||
+  w.mozRequestAnimationFrame;
+
+function Reset() {
+  location.reload();
+}
+
+function initializeClock(id, endtime) {
+  var timer = document.getElementById(id);
+
+  function updateClock() {
+    var t = getTimeRemaining(endtime);
+
+    secondsSpan.innerHTML = ("0" + t.seconds).slice(-2);
+
+    if (t.total <= 0) {
+      secondsSpan.innerHTML = "0";
+    }
+  }
+
+  updateClock();
+  timeinterval = setInterval(updateClock, 1000);
+}
+
+function stopTimer(clock) {
+  clearInterval(clock);
+  // secondsSpan.innerHTML = "0";
+}
+
+function getTimeRemaining(endtime) {
+  var t = Date.parse(endtime) - Date.parse(new Date());
+  var seconds = Math.floor((t / 1000) % 60);
+
+  return {
+    total: t,
+
+    seconds: seconds
+  };
+}
+
+function closeForm(element) {
+  document.getElementById(element).style.display = "none";
+}
+
+function submitName() {
+  let userInputName = document.getElementById("nameInput").value;
+  var playerName = document.getElementById("playerName");
+
+  if (userInputName === "") {
+    playerName.innerHTML = "<p style='color:red;'>Calling your name...</p>";
+    return;
+  }
+
+  let player = document.getElementById("playerName");
+  player.innerHTML = "Hello " + userInputName + "!";
+  closeForm("myForm");
+
+  playAudio();
+  var deadline = new Date(Date.parse(new Date()) + 1 * 1 * 1 * maxTime * 1000);
+  initializeClock("remain-time", deadline);
+  main();
+}
+
+let submitButton = document.getElementById("submitBtn");
+submitButton.addEventListener("click", submitName);
 
 // Let's play this game!
+
+backgroundMusic = document.getElementById("myBgMusic");
+
+function playAudio() {
+  myBgMusic.play();
+}
+let volOnBtn = document.getElementById("volOn");
+volOnBtn.addEventListener("click", playAudio);
+
+function pauseAudio() {
+  myBgMusic.pause();
+}
+let volOffBtn = document.getElementById("volOff");
+volOffBtn.addEventListener("click", pauseAudio);
+
+function toggle() {

```

Comment:
Cool! I really like your use of  toggle functions to hide and display elements... i wish i did this!